.github/workflows/ci.yml

```
@@ -0,0 +1,68 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - "master"
+  pull_request:
+    branches:
+      - "master"
+  schedule:
+    # Run a cron job once daily, weekdays only
+    - cron: "0 3 1,15 * *"
+
+jobs:
+  test:
+    name: Pytest
+    runs-on: ${{ matrix.cfg.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        cfg:
+          #- os: ubuntu-latest
+          #  python-version: "3.6"
+          #- os: ubuntu-latest
+          #  python-version: "3.7"
+          - os: ubuntu-latest
+            python-version: "3.8"
+          #- os: ubuntu-latest
+          #  python-version: "3.9"
+          #- os: macos-latest
+          #  python-version: "3.6"
+          #- os: windows-latest
+          #  python-version: "3.6"
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - uses: conda-incubator/setup-miniconda@v2
+        with:
+          python-version: ${{ matrix.cfg.python-version }}
+          activate-environment: kinfraglib
+          channel-priority: true
+          environment-file: environment.yml
+          auto-activate-base: false
+
+      - name: Additional info about the build
+        shell: bash
+        run: |
+          uname -a
+          df -h
+          ulimit -a
+
+      - name: Environment Information
+        shell: bash -l {0}
+        run: |
+          conda info --all
+          conda list
+
+      - name: Run tests
+        shell: bash -l {0}
+        run: |
+          echo "Download combinatorial library from zenodo..."
+          wget -q -O data/combinatorial_library/combinatorial_library.tar.bz2 https://zenodo.org/record/3956580/files/combinatorial_library.tar.bz2?download=1
+          echo "Decompress combinatorial library..."
+          tar -xvf data/combinatorial_library/combinatorial_library.tar.bz2 -C data/combinatorial_library/
+          echo "Run tests..."
+          PYTEST_ARGS="--nbval-lax --current-env --dist loadscope --numprocesses 2"
```

```suggestion
          PYTEST_ARGS="--nbval-lax --current-env --nbval-cell-timeout=900"
```

Debug without paralellism for now (`--dist loadscope --numprocesses 2`), but put it back later if you can.

 --current-env         Force test execution to use a python kernel in the same
                        enviornment that py.test was launched from.